src/dataSource.js

```diff
@@ -90,84 +87,201 @@ class DataSource {
    *
    * @param {number} row Physical row index.
-   * @param {number} column Visual column index.
-   * @returns {*}
+   * @param {number|string} column Property name / physical column index.
+   * @param {*} value The value to be set at the provided coordinates.
    */
-  getAtCell(row, column) {
-    let result = null;
+  setAtCell(row, column, value) {
+    if (row >= this.countRows() || column >= this.countFirstRowKeys()) {
+      // Not enough rows and/or columns.
+      return;
+    }
 
-    const modifyRowData = this.hot.runHooks('modifyRowData', row);
+    if (this.hot.hasHook('modifySourceData')) {
+      const valueHolder = createObjectPropListener(value);
 
-    const dataRow = isNaN(modifyRowData) ? modifyRowData : this.data[row];
+      this.hot.runHooks('modifySourceData', row, this.propToCol(column), valueHolder, 'set');
 
-    if (dataRow) {
-      const prop = this.colToProp(column);
+      if (valueHolder.isTouched()) {
+        value = valueHolder.value;
+      }
+    }
 
-      if (typeof prop === 'string') {
-        result = getProperty(dataRow, prop);
+    if (!Number.isInteger(column)) {
+      // column argument is the prop name
+      setProperty(this.data[row], column, value);
 
-      } else if (typeof prop === 'function') {
-        result = prop(this.data.slice(row, row + 1)[0]);
+    } else {
+      this.data[row][column] = value;
+    }
+  }
+
+  /**
+   * Get data from the source data set using the physical indexes.
+   *
+   * @private
+   * @param {number} row Physical row index.
+   * @param {string|number|Function} column Physical column index / property / function.
+   * @param {Array|object} dataRow A representation of a data row.
+   * @returns {*} Value at the provided coordinates.
+   */
+  getAtPhysicalCell(row, column, dataRow) {
+    let result = null;
+
+    if (dataRow) {
+      if (typeof column === 'string') {
+        result = getProperty(dataRow, column);
+
+      } else if (typeof column === 'function') {
+        result = column(dataRow);
 
       } else {
-        result = dataRow[prop];
+        result = dataRow[column];
+      }
+    }
+
+    if (this.hot.hasHook('modifySourceData')) {
+      const valueHolder = createObjectPropListener(result);
+
+      this.hot.runHooks('modifySourceData', row, this.colToProp(column), valueHolder, 'get');
+
+      if (valueHolder.isTouched()) {
+        result = valueHolder.value;
       }
     }
 
     return result;
   }
 
+  /**
+   * Returns a single value from the data.
+   *
+   * @param {number} row Physical row index.
+   * @param {number} column Visual column index.
+   * @returns {*}
+   */
+  getAtCell(row, column) {
+    let modifyRowData = null;
+
+    if (this.hot.hasHook('modifyRowData')) {
+      modifyRowData = this.hot.runHooks('modifyRowData', row);
+    }
+
+    const dataRow = modifyRowData !== null && !Number.isInteger(modifyRowData) ? modifyRowData : this.data[row];
```

Comment:
`modifyRowData` is handled differently here than in `getAtRow` method. Maybe we should extract this to a common private handler so we have the result from the hook always processed the same way?